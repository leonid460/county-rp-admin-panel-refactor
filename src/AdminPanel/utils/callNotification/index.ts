import {eventEmitter} from 'AdminPanel/eventEmitter';
import {
  notificationEvent,
  NotificationMsg,
} from 'AdminPanel/components/molecules/Notification';

export function callNotification(data: NotificationMsg) {
  eventEmitter.emit<NotificationMsg>(notificationEvent, data);
}
