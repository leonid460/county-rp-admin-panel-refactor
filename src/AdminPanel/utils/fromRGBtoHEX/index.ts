export function fromRGBtoHEX(R: number, G: number, B: number) {
  let hexColor = '';

  [R, G, B].forEach(num => {
    const hexNum = num.toString(16);

    hexColor += hexNum.length > 1 ? hexNum : `0${hexNum}`;
  });

  return '#'+hexColor;
}
