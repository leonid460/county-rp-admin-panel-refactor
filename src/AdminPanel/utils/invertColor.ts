export function invertColor(hexColor: string) {
  return (Number(`0x1${hexColor}`) ^ 0xffffff)
    .toString(16)
    .substr(1)
    .toUpperCase();
}
