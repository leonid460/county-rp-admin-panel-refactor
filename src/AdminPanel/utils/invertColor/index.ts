export function invertColor(hexColor: string) {
  if (hexColor[0] !== '#') {
    throw new Error('Wrong hex color string format (missing "#" at the start): '+ hexColor);
  }

  const cuttedHex = hexColor.slice(1);

  return '#'+(Number(`0x1${cuttedHex}`) ^ 0xFFFFFF).toString(16).substr(1);
}
