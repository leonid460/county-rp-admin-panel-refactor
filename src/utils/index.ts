export * from './fromRGBtoHEX';
export * from './invertColor';
export * from './makeColorFromString';
export * from './callNotification';
