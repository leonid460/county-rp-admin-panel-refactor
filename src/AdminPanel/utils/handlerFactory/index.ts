import {callNotification} from 'AdminPanel/utils/callNotification';

export function handlerFactory(
  service: () => Promise<number>,
  successCallback: () => void
) {
  return async () => {
    try {
      const fetchResult = await service();

      if (fetchResult === 0) {
        successCallback();
        callNotification({
          header: '',
          content: 'изменения сохранены',
        });
      }
    } catch (error) {
      handlerFetchError(error);
      console.dir(error);
    }
  };
}

export function handlerFetchError(error: Error) {
  callNotification({
    header: 'Ошибка',
    content: error.message,
    type: 'error',
  });
}
