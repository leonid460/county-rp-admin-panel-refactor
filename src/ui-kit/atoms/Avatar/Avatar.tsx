import React from 'react';
import { AvatarContainer } from './styled';
import { IAvatarProps } from './types';
import { invertColor, makeColorFromString } from 'utils';

export const Avatar: React.FC<IAvatarProps> = ({ className, children }) => {
  const text = children.toString();
  const backgroundColor = makeColorFromString(text);
  const textColor = invertColor(backgroundColor);

  return (
    <AvatarContainer className={className} textColor={textColor} backgroundColor={backgroundColor}>
      {text.slice(0, 2).toUpperCase()}
    </AvatarContainer>
  );
};
