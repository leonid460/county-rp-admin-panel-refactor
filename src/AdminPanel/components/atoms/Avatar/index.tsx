import React from 'react';
import styled from 'styled-components';
import {makeColorFromString} from 'AdminPanel/utils/makeColorFromString';
import {invertColor} from 'AdminPanel/utils/invertColor';


type AvatarProps = {
  textColor: string;
  backgroundColor: string;
  children: React.ReactText;
}

const TopMenuLogo = styled.div<AvatarProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  
  min-width: 50px;
  min-height: 50px;
  max-width: 50px;
  max-height: 50px;
  border-radius: 50%;
  margin: 0 15px;
  font-weight: bold;
  font-size: 24px;

  background: ${props => props.backgroundColor};
  color: ${props => props.textColor};
`;

export default ({className, children}: {
  className?: string;
  children: React.ReactText;
}) => {
  const text = children.toString();
  const backgroundColor = makeColorFromString(text);
  const textColor = invertColor(backgroundColor);

  return (
    <TopMenuLogo className={className} textColor={textColor} backgroundColor={backgroundColor}>
      {text.slice(0, 2).toUpperCase()}
    </TopMenuLogo>
  );
};
