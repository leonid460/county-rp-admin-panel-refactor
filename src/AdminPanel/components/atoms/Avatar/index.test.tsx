import React from 'react';
import Avatar from './index';
import {makeColorFromString} from 'AdminPanel/utils/makeColorFromString';
import {invertColor} from 'AdminPanel/utils/invertColor';
import 'jest-styled-components';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';


describe('testing <Avatar />', () => {
  it('renders without crashing', () => {
    shallow(<Avatar>Test</Avatar>);
  });

  it('matches snapshot', () => {
    const component = <Avatar>Test</Avatar>;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('makes colors correctly', () => {
    const text = 'Test';
    const backgroundColor = makeColorFromString(text);
    const textColor = invertColor(backgroundColor);
    const component = shallow(<Avatar>{text}</Avatar>);
    expect(component).toHaveStyleRule('background', backgroundColor);
    expect(component).toHaveStyleRule('color', textColor);
  });
});
