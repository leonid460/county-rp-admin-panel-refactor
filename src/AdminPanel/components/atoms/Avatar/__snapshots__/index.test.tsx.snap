// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Avatar /> matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 50px;
  min-height: 50px;
  max-width: 50px;
  max-height: 50px;
  border-radius: 50%;
  margin: 0 15px;
  font-weight: bold;
  font-size: 24px;
  background: #27b8b2;
  color: #d8474d;
}

<div
  className="c0"
>
  TE
</div>
`;
